src/app/pages/payment/pin-payment.component.ts

```diff
@@ -0,0 +1,102 @@
+import { validateInput } from 'src/app/validators/payment-validators';
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, Validators, FormControl } from '@angular/forms';
+import { PaymentService } from 'src/app/services/payment-service';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { Router } from '@angular/router';
+import { ToastrService } from 'ngx-toastr';
+import { Location } from '@angular/common';
+import { Title } from '@angular/platform-browser';
+
+@Component({
+  selector: 'app-pin-payment',
+  templateUrl: './pin-payment.component.html',
+  styleUrls: ['./payment.component.scss']
+})
+export class PinPaymentComponent implements OnInit {
+  validYears: number[];
+  validMonths: string[] = [
+    '01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'
+  ];
+  cardForm = new FormGroup({
+    cardNumber: new FormControl('', Validators.required),
+    securityNumber: new FormControl('', Validators.required),
+    cardExpiry: new FormGroup({
+      expiryMonth: new FormControl(
+        'MM', [Validators.required, validateInput('MM')]
+        ),
+      expiryYear: new FormControl(
+        'YYYY', [Validators.required, validateInput('YYYY')]
+        )
+    }),
+    amount: new FormControl('', [Validators.required, Validators.min(0)]),
+    pin: new FormControl('', Validators.required),
+    purpose: new FormControl('Purpose', [Validators.required, validateInput('Purpose')]),
+    saveCard: new FormControl(false)
+  });
+
+  constructor(
+    private paymentService: PaymentService,
+    private router: Router,
+    private spinner: NgxSpinnerService,
+    private toastr: ToastrService,
+    private location: Location,
+    private titleService: Title
+    ) {this.titleService.setTitle('Card & Pin Deposit'); }
+
+  ngOnInit(): void {
+    this.generateValidYears();
+  }
+
+  generateValidYears() {
+    const yearList: number[] = [];
+    const now = new Date();
+    const currentYear = now.getFullYear();
+    const lastYear = currentYear + 10;
```

Comment:
Hey @Victoradukwu, this variable name is quite misleading, could we rewrite it to something that's more straight forward.

Reply:
Fixed, @CryceTruly